### PR TITLE
Remove unnecessary config marshalling wherever possible

### PR DESCRIPTION
**Pull request by jaronoff97**
**Description:** <Describe what has changed.>

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #issue-number

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: internal/manifests/collector/adapters/config_to_probe_test.go [hunk 1]
```diff
@@ -131,57 +137,40 @@ func TestConfigToProbeShouldErrorIf(t *testing.T) {
 service:
   extensions: [health_check]`,
 			expectedErr: errNoExtensionHealthCheck,
-		}, {
```
Comment (jaronoff97) [inline internal/manifests/collector/adapters/config_to_probe_test.go:134]: some of these tests are no longer applicable!

File: apis/v1beta1/config.go [hunk 2]
```diff
@@ -98,14 +98,56 @@ func (c Config) Yaml() (string, error) {
 	return buf.String(), nil
 }
 
+// MetricsConfig comes from the collector
+type MetricsConfig struct {
```
Comment (pavolloffay) [inline apis/v1beta1/config.go:102]: Could this be introduced in a separate PR?

File: apis/v1beta1/config.go [hunk 3]
```diff
@@ -98,14 +98,56 @@ func (c Config) Yaml() (string, error) {
...
+	// Resource specifies user-defined attributes to include with all emitted telemetry.
+	// Note that some attributes are added automatically (e.g. service.version) even
+	// if they are not specified here. In order to suppress such attributes the
+	// attribute must be specified in this map with null YAML value (nil string pointer).
+	Resource map[string]*string `json:"resource,omitempty" yaml:"resource,omitempty"`
+}
+
 type Service struct {
-	Extensions *[]string `json:"extensions,omitempty" yaml:"extensions,omitempty"`
+	Extensions []string `json:"extensions,omitempty" yaml:"extensions,omitempty"`
```
Comment (pavolloffay) [inline apis/v1beta1/config.go:125]: IIRC this will enforce marshalling of empty array for extensions. Is this necessary?

File: apis/v1beta1/config.go [hunk 4]
```diff
@@ -98,14 +98,56 @@ func (c Config) Yaml() (string, error) {
...
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Telemetry *AnyConfig `json:"telemetry,omitempty" yaml:"telemetry,omitempty"`
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Pipelines AnyConfig `json:"pipelines" yaml:"pipelines"`
 }
 
+// GetTelemetry serves as a helper function to access the fields we care about in the underlying telemetry struct.
+// This exists to avoid needing to worry extra fields in the telemetry struct.
+func (s *Service) GetTelemetry() *Telemetry {
```
Comment (pavolloffay) [inline apis/v1beta1/config.go:134]: could we add a test for this? Getting a telemetry from an existing yaml?

**Comment by jaronoff97** [comment]
closing in favor of #2857
